lib/src/segy.c

```diff
@@ -1846,6 +1846,84 @@ int segy_traces( segy_datasource* ds,
     return SEGY_OK;
 }
 
+/* Allows only int2 and uint2. */
+static int field_data_to_scalar(
```

Comment:
Do we expect this method to become more generalised? If not, should the restriction (only be usable for 2 byte ints) be reflected in the method's name?

Reply:
No, I think this method is generalized enough.
No, I don't think restriction should be in the method's name. There are 4 scalars present in the specification, all 2 bytes.
PO decided that we should accept scalars only of this size.
Changing the name would imply other options are possible.

If someone complains in the future - then we shall look at it again.